framework/quota/test/org/apache/cloudstack/quota/QuotaStatementTest.java

```diff
@@ -0,0 +1,248 @@
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.cloudstack.quota;
+
+import com.cloud.user.AccountVO;
+import com.cloud.user.dao.AccountDao;
+import com.cloud.utils.db.TransactionLegacy;
+import junit.framework.TestCase;
+import org.apache.cloudstack.framework.config.dao.ConfigurationDao;
+import org.apache.cloudstack.quota.QuotaStatementImpl.STATEMENT_PERIODS;
+import org.apache.cloudstack.quota.dao.QuotaAccountDao;
+import org.apache.cloudstack.quota.dao.QuotaUsageDao;
+import org.apache.cloudstack.quota.vo.QuotaAccountVO;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import javax.naming.ConfigurationException;
+
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QuotaStatementTest extends TestCase {
+
+    @Mock
+    AccountDao accountDao;
+    @Mock
+    QuotaAccountDao quotaAcc;
+    @Mock
+    ConfigurationDao configDao;
+    @Mock
+    QuotaUsageDao quotaUsage;
+    @Mock
+    QuotaAlertManager emailQuotaAlert;
+
+    @Spy
+    QuotaStatementImpl quotaStatement = new QuotaStatementImpl();
+
+    private void injectMockToField(Object mock, String fieldName) throws NoSuchFieldException, IllegalAccessException {
+        Field f = QuotaStatementImpl.class.getDeclaredField(fieldName);
+        f.setAccessible(true);
+        f.set(quotaStatement, mock);
+    }
+
+    @Before
+    public void setup() throws IllegalAccessException, NoSuchFieldException, ConfigurationException {
+        // Dummy transaction stack setup
+        TransactionLegacy.open("QuotaStatementImplTest");
+
+        injectMockToField(accountDao, "_accountDao");
+        injectMockToField(quotaAcc, "_quotaAcc");
+        injectMockToField(configDao, "_configDao");
+        injectMockToField(quotaUsage, "_quotaUsage");
+        injectMockToField(emailQuotaAlert, "_quotaAlert");
+    }
+
+    @Test
+    public void testStartStop() {
+        try {
+            quotaStatement.start(); // expected to fail as pid is not available
+        } catch (NumberFormatException ignored) {
+        }
+        assertTrue(quotaStatement.stop());
```

Comment:
Why would we test the ``stop`` operation in a test where we expect the ``start`` operation to fail?